Sources/LanguageServerProtocol/Scheme.swift

```diff
@@ -0,0 +1,22 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+public struct Scheme: ResponseType, Hashable {
+  /// Unique identifier for this scheme.
+  public let identifier: String
+
+  /// Build configuration to use (e.g. "debug" or "release").
+  public let configuration: String?
+  
+  /// Targets in the given scheme.
+  public let targets: [String?]
```

Comment:
This would need to be a BuildTargetIdentifier, which is going to be in the BuildServerProtocol target instead. Bit tricky.

Reply:
That would require some major refactor, how many things do `BuildServerProtocol` rely on language server protocol? Is it just the underlying message protocol to get the free parsing and request/response handling?